AVRIoT.X/mcc_generated_files/cloud/wifi_service.c

```diff
@@ -206,7 +206,7 @@ bool wifi_connectToAp(uint8_t passed_wifi_creds)
 	
 	if(passed_wifi_creds == NEW_CREDENTIALS)
 	{
-		e=m2m_wifi_connect((char *)ssid, sizeof(ssid), atoi((char*)authType), (char *)pass, M2M_WIFI_CH_ALL);
```

Comment:
Casting is not required (the ssid, pass and authType buffers are already of type  (char*) 
More importantly the second parameter passed to m2m_wifi_connect is supposed to be the length of the string not the size of the buffer. The use of sizeof could  (and currently does)  still work, but prevents us from allocating buffers of the correct size (32+1 and 64+1 respectively to leave room for the string terminator '\0') since the function does perform a check of the input parameters against the max valid values (32 and 64).